app/src/main/java/com/github/sdpteam15/polyevents/view/activity/admin/EventEditManagementActivity.kt

```diff
@@ -0,0 +1,197 @@
+import android.transition.TransitionManager
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.widget.Button
+import android.widget.LinearLayout
+import android.widget.PopupWindow
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.RecyclerView
+import com.github.sdpteam15.polyevents.R
+import com.github.sdpteam15.polyevents.helper.HelperFunctions
+import com.github.sdpteam15.polyevents.model.database.remote.Database
+import com.github.sdpteam15.polyevents.model.database.remote.DatabaseConstant
+import com.github.sdpteam15.polyevents.model.entity.Event
+import com.github.sdpteam15.polyevents.model.observable.ObservableList
+import com.github.sdpteam15.polyevents.model.observable.ObservableMap
+import com.github.sdpteam15.polyevents.view.adapter.EventEditAdminAdapter
+import com.github.sdpteam15.polyevents.view.fragments.admin.EventEditDifferenceFragment
+
+class EventEditManagementActivity : AppCompatActivity() {
+    private lateinit var recyclerView: RecyclerView
+    private val eventEdits = ObservableList<Event>()
+    private val origEvents = ObservableMap<String, Event>()
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_event_edit_management)
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+
+        recyclerView = findViewById(R.id.id_recycler_event_edits)
+
+
+        val eventList = ObservableList<Event>()
+
+        eventList.observeAdd(this) {
+            val event = it.value
+            origEvents[event.eventId!!] = event
+        }
+
+        Database.currentDatabase.eventDatabase!!.getEvents(null, null, eventList)
+            .observeOnce(this) {
+                if (!it.value) {
+                    HelperFunctions.showToast("Failed to get the list of all events", this)
+                    finish()
+                } else {
+                    getEventEdit()
+                }
+            }
+    }
+
+    private fun getEventEdit() {
+        //Wait until we have both requests accepted from the database to show the material requests
+        Database.currentDatabase.eventDatabase!!.getEventEdits(
+            { collection ->
+                collection.orderBy(DatabaseConstant.EventEditConstant.EVENT_EDIT_STATUS.value)
+            },
+            eventEdits,
+        ).observeOnce(this) {
+            if (!it.value) {
+                HelperFunctions.showToast("Failed to get the list of material requests", this)
+                finish()
+            } else {
+                recyclerView.adapter =
+                    EventEditAdminAdapter(
+                        this,
+                        this,
+                        eventEdits,
+                        origEvents,
+                        acceptEventEditRequest,
+                        declineEventEdit,
+                        seeEventEdit
```

Comment:
you can put the adapter outside, just after "recyclerView = findViewById(R.id.id_recycler_event_edits)". The recycler is automaticaly updated, because you are already calling a notifyDatasetChanged() on a modification of your observable lists in the adapter

Reply:
Yes, you're correct !